### ISSUE TITLE
Sequential I/O vs  Distributed I/O test results differ

### ISSUE THREAD

**Issue by smithsg84** [parflow/parflow#206]
We have noticed that building ParFlow with sequential I/O vs. distributed I/O produces different results when CTest runs the set of 155 build tests. Specifically, two test cases, llnl-perm and washita, both fail when ParFlow is compiled with distributed IO (default) while they both pass when ParFlow is compiled with sequential IO (DPARFLOW_AMPS_SEQUENTIAL_IO=True). 

To reproduce this: Compile ParFlow with -DPARFLOW_AMPS_SEQUENTIAL_IO=True, then cd to the build directory and run make test. Compare the results when compiling without setting the above flag and note that two tests fail for distributed I/O.

I have been able to identify that parflow/examples/llnl-perm.tcl needs a line to undistribute the llnl.out.perm_x.pfb file prior to attempting to load the file. I believe this proposed solution will align the test results for sequential I/O and distributed I/O.

The washita test is a bit more complicated as the process depends on the compilation method. For distributed I/O, the inputs need to be distributed prior to running the simulation, then the outputs need to be undistributed prior to testing them for accuracy. 
However, for the case that ParFlow was compiled with sequential I/O, the inputs should not be distributed or else the test will fail to run. The undistribution of the outputs appears to have no effect on the tests passing/failing in this case. 

I believe it is desirable to have the test results be consistent regardless of the value of DPARFLOW_AMPS_SEQUENTIAL_IO at compilation time. Prior to suggesting any changes, I wanted to ask, is there a known reason to leave these tests in their current state?

Thank you for your time, and please let me know if I have overlooked something or made any incorrect assumptions.

-Ahmad Rezaii

**Comment by smithsg84** [comment]
Thanks for bug report. 

Results should be the same so this is not intended or correct behavior.

It is supposed to be the case that distribute/undistribute  should work no matter what the I/O model to keep the scripts the same independent of I/O model.

**Comment by smithsg84** [comment]
Fix that works locally for me is on branch.

https://github.com/parflow/parflow/tree/bugfix/206-sequential-distributed-io

**Comment by arezaii** [comment]
I can confirm both sequential and distributed work with the fixes here. Would there be any harm to undistribute the CLM outputs for all timesteps instead of just the 12th timestep? Something like: 
```
set StartTime [expr int([pfget TimingInfo.StartTime])]
set StopTime [expr int([pfget TimingInfo.StopTime])]

set clm [list "eflx_lh_tot" "qflx_evap_soi" "swe_out" "eflx_lwrad_out" "qflx_evap_tot" "t_grnd" "eflx_sh_tot" "qflx_evap_veg" "t_soil" "eflx_soil_grnd" "qflx_infl" "qflx_evap_grnd" "qflx_tran_veg" ]
 for {set i $StartTime} { $i <= $StopTime } {incr i} { 
  set step [format "%05d" $i]
    foreach clm_var $clm {
        pfundist $runname.out.$clm_var.$step.pfb
    }
}
```
This way the Outputs directory will only contain .pfb files and no .00000 files for the user to undistribute manually.

**Comment by smithsg84** [comment]
Thanks for reviewing.   That sounds good.  I'll add the undist for everything and update.

**Comment by arezaii** [comment]
Tested the master branch after merging. All 155 tests pass in both sequential and distributed build configurations. Thank you for the help with this!

**Comment by smithsg84** [comment]
This was fixed in PR #207
